<details><summary>Failed to submit graph for 3889aa675da2572c3f8fd4b728f08e052ba8bd40
</summary>Message:
Failed to run command: 'pbp generate from-feedstock --output /tmp/tmp6un4ojb1/pylint-feedstock.yaml --custom-metadata-file /tmp/tmp6un4ojb1/custom_metadata.yaml --release-name pylint-feedstock-28-3889aa675da2572c3f8fd4b728f08e052ba8bd40 pylint-feedstock/recipe', with error: 'INFO:root:Found feedstock pylint-feedstock at /tmp/tmp6un4ojb1/pylint-feedstock/recipe
INFO:root:Build/Test graph from pylint-feedstock
Created by sbx_user1051 on 2026-01-08 13:19:35
Command line: /opt/conda/envs/webhook-listener/bin/pbp generate from-feedstock --output /tmp/tmp6un4ojb1/pylint-feedstock.yaml --custom-metadata-file /tmp/tmp6un4ojb1/custom_metadata.yaml --release-name pylint-feedstock-28-3889aa675da2572c3f8fd4b728f08e052ba8bd40 pylint-feedstock/recipe

INFO:root:Found abs.yaml in /tmp/tmp6un4ojb1/pylint-feedstock
INFO:pbpcli.utils.common:Setting env var: ANACONDA_ROCKET_ENABLE_PY314 = True.
INFO:root:Creating graph from feedstock at: pylint-feedstock
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.12 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 3 unchanging vars:(455), changing vars (3): ['blas_impl', 'numpy', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
WARNING: Number of parsed outputs does not match detected raw metadata blocks. Identified output block may be wrong! If you are using Jinja conditionals to include or exclude outputs, consider using `skip: true  # [condition]` instead.
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
INFO:pbpcli.utils.common:Unsetting env var: ANACONDA_ROCKET_ENABLE_PY314
Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/bin/pbp", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 413, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 624, in from_feedstock
    build_configs, test_configs = expand_build_matrix(feedstock.recipe_dir_path, src_channels)
                                  ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 79, in expand_build_matrix
    _targets, build_configs, test_configs = variants_for_feedstock_and_platform(
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        recipe_dir_path, platform, src_channels
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/utils/expand_variants.py", line 544, in variants_for_feedstock_and_platform
    metadata_tuples = api.render(
        parsed.recipe,
    ...<2 lines>...
        variants=None,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/api.py", line 68, in render
    return render_metadata_tuples(
        metadata_tuples,
    ...<3 lines>...
        bypass_env_check=bypass_env_check,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 1097, in render_metadata_tuples
    for od, om in meta.get_output_metadata_set(
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        permit_undefined_jinja=not finalize,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        bypass_env_check=bypass_env_check,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    ):
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 2781, in get_output_metadata_set
    conda_packages = finalize_outputs_pass(
        ref_metadata,
    ...<3 lines>...
        bypass_env_check=bypass_env_check,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 1077, in finalize_outputs_pass
    fm = finalize_metadata(
        om,
        parent_metadata=parent_metadata,
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 722, in finalize_metadata
    build_unsat, host_unsat = add_upstream_pins(
                              ~~~~~~~~~~~~~~~~~^
        m, permit_unsatisfiable_variants, exclude_pattern, extra_specs
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 507, in add_upstream_pins
    host_deps, host_unsat, extra_run_specs_from_host = _read_upstream_pin_files(
                                                       ~~~~~~~~~~~~~~~~~~~~~~~~^
        m, "host", permit_unsatisfiable_variants, exclude_pattern, extra_specs
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 463, in _read_upstream_pin_files
    extra_run_specs = get_upstream_pins(m, precs, env)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 433, in get_upstream_pins
    loc, dist = execute_download_actions(
                ~~~~~~~~~~~~~~~~~~~~~~~~^
        m,
        ^^
    ...<2 lines>...
        package_subset=[prec],
        ^^^^^^^^^^^^^^^^^^^^^^
    )[prec]
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 396, in execute_download_actions
    pfe.execute()
    ~~~~~~~~~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda/core/package_cache_data.py", line 918, in execute
    raise CondaMultiError(not_cancelled)
conda.CondaMultiError: ('Connection broken: IncompleteRead(6028438 bytes read, 30906666 more expected)', IncompleteRead(6028438 bytes read, 30906666 more expected))
('Connection broken: IncompleteRead(6028438 bytes read, 30906666 more expected)', IncompleteRead(6028438 bytes read, 30906666 more expected))

'
aws lambda request id: 7c41225d-1272-5fb8-8022-4384a37d2a00
[View in Grafana](https://grafana.anacondaconnect.com/explore?schemaVersion=1&panes=%7B%2286g%22:%7B%22datasource%22:%22loki-central%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7B__aws_cloudwatch_log_group%3D%5C%22%2Faws%2Flambda%2Fpbp-graph-submitter-prod%5C%22%7D%20%7C%20json%20%7C%20requestId%20%3D%20%607c41225d-1272-5fb8-8022-4384a37d2a00%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-central%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22columns%22:%7B%220%22:%22Time%22,%221%22:%22Line%22%7D,%22visualisationType%22:%22table%22,%22labelFieldName%22:%22labels%22%7D%7D%7D%7D&orgId=1)</details>

--- a/abs.yaml
+++ b/abs.yaml
@@ -1,2 +1,2 @@
 build_env_vars:
-  ANACONDA_ROCKET_ENABLE_PY313 : yes
+  ANACONDA_ROCKET_ENABLE_PY314 : yes

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -10,7 +10,7 @@ source:
   sha256: 26698de19941363037e2937d3db9ed94fb3303fdadf7d98847875345a8bb6b05
 
 build:
-  number: 0
+  number: 1
   skip: true  # [py<39]
   script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation --ignore-installed --no-cache-dir -vvv
   entry_points:
@@ -41,7 +41,18 @@ requirements:
     - gitpython >3
     - pyenchant >=3.2,<4
 
+# >   os.remove(tempfile_name)
+# E   PermissionError: [Errno 1] Operation not permitted: '/tmp/CONFIG'
+{% set deselect_tests = " --deselect=tests/config/pylint_config/test_pylint_config_generate.py::test_writing_to_output_file" %}
+{% set deselect_tests = deselect_tests + " --deselect=tests/config/test_find_default_config_files.py::test_pylintrc" %}  # [win]
+{% set deselect_tests = deselect_tests + " --deselect=tests/config/test_find_default_config_files.py::test_pyproject_toml_parentdir" %}  # [win]
+{% set deselect_tests = deselect_tests + " --deselect=tests/lint/unittest_lint.py::test_one_arg" %}  # [win]
+{% set deselect_tests = deselect_tests + " --deselect=tests/lint/unittest_lint.py::test_two_similar_args" %}  # [win]
+{% set deselect_tests = deselect_tests + " --deselect=tests/lint/unittest_lint.py::test_more_args" %}  # [win]
 test:
+  source_files:
+    - pyproject.toml
+    - tests
   imports:
     - pylint
     - pylint.checkers
@@ -50,35 +61,32 @@ test:
     - pylint.reporters
     - pylint.reporters.ureports
     - pylint.utils
+  commands:
+    - pip check
+    # Hardcoded test output does not work in our test environment due to pylint being seen as a local package. Updating
+    # the function output was tested; however, dependant tests continue to fail. Disabling for now.
+    # https://github.com/pylint-dev/pylint/blob/v3.2.7/tests/functional/w/wrong_import_order.txt
+    # Disabled tests for the python-314, won`t rebuild deps only for the tests
+    - pytest -v tests -k "not (test_functional[wrong_import_order] or test_compare[both_empty] or test_compare[no_change] or test_compare[message_changed] or test_compare_batched or test_truncated_compare)" {{ deselect_tests }}  # [py<314]
+    - pylint --help
+    - pyreverse --help
+    - symilar --help
+    # Has no help option.
+    # Running without arguments is an error.
+    # So just check that it exists.
+    - type pylint-config  # [unix]
+    - where pylint-config  # [win]
   requires:
     - git
     - gitpython
     - pip
     - py >=1.11.0,<1.12
     - pytest >=7.4.0,<8
-    - pytest-benchmark >=4.0,<5
+    - pytest-benchmark >=4.0,<5  # [py<314]
     - pytest-timeout >=2,<3
     - pytest-xdist >=3.6,<4
     - requests
     - typing-extensions >=4.12,<5
-  source_files:
-    - pyproject.toml
-    - tests
-  commands:
-    - pip check
-    # Hardcoded test output does not work in our test environment due to pylint being seen as a local package. Updating
-    # the function output was tested; however, dependant tests continue to fail. Disabling for now.
-    # https://github.com/pylint-dev/pylint/blob/v3.2.7/tests/functional/w/wrong_import_order.txt
-    - pytest -v tests -k "not (test_functional[wrong_import_order] or test_compare[both_empty] or
-      test_compare[no_change] or test_compare[message_changed] or test_compare_batched or test_truncated_compare)"
-    - pylint --help
-    - pyreverse --help
-    - symilar --help    
-    # Has no help option.
-    # Running without arguments is an error.
-    # So just check that it exists.
-    - type pylint-config  # [unix]
-    - where pylint-config  # [win]
 
 about:
   home: https://www.pylint.org

